runtime/src/iree/hal/drivers/hip/pending_queue_actions.c

```diff
@@ -802,14 +991,56 @@ static iree_status_t iree_hal_hip_pending_queue_actions_issue_execution(
   iree_slim_mutex_unlock(
       &action->owning_actions->working_area.pending_work_items_count_mutex);
 
-  // Now launch a host function on the callback stream to advance the semaphore
-  // timeline.
-  IREE_HIP_RETURN_AND_END_ZONE_IF_ERROR(
-      z0, symbols,
-      hipLaunchHostFunc(action->callback_hip_stream,
-                        iree_hal_hip_execution_device_signal_host_callback,
-                        action),
-      "hipLaunchHostFunc");
+  iree_hal_hip_atomic_slist_completion_t* entry = NULL;
+  // TODO: avoid host allocator malloc; use some pool for the allocation.
+  iree_status_t status = iree_allocator_malloc(
+      action->owning_actions->host_allocator, sizeof(*entry), (void**)&entry);
+
+  if (IREE_UNLIKELY(!iree_status_is_ok(status))) {
+    // Release all actions in the ready list to avoid leaking.
```

Comment:
Nothing released here?

Reply:
Oh good catch that's a no longer relevant comment.